editor/plugins/tiles/atlas_merging_dialog.cpp

```diff
@@ -69,20 +69,26 @@ void AtlasMergingDialog::_generate_merged(Vector<Ref<TileSetAtlasSource>> p_atla
 				Vector2i tile_id = atlas_source->get_tile_id(tile_index);
 				atlas_size = atlas_size.max(tile_id + atlas_source->get_tile_size_in_atlas(tile_id));
 
-				Rect2i new_tile_rect_in_altas = Rect2i(atlas_offset + tile_id, atlas_source->get_tile_size_in_atlas(tile_id));
+				Rect2i new_tile_rect_in_atlas = Rect2i(atlas_offset + tile_id, atlas_source->get_tile_size_in_atlas(tile_id));
 
 				// Copy the texture.
+				int animation_columns = atlas_source->get_tile_animation_columns(tile_id);
+				Vector2i animation_separation = atlas_source->get_tile_animation_separation(tile_id);
 				for (int frame = 0; frame < atlas_source->get_tile_animation_frames_count(tile_id); frame++) {
 					Rect2i src_rect = atlas_source->get_tile_texture_region(tile_id, frame);
-					Rect2 dst_rect_wide = Rect2i(new_tile_rect_in_altas.position * new_texture_region_size, new_tile_rect_in_altas.size * new_texture_region_size);
+					Vector2i frame_coords = tile_id + (src_rect.size + animation_separation) * ((animation_columns > 0) ? Vector2i(frame % animation_columns, frame / animation_columns) : Vector2i(frame, 0));
+					Vector2i extra = (frame > 0 ? frame_coords : Vector2i(0, 0));
+					atlas_size.x += extra.x;
+					Vector2i newPosition = new_tile_rect_in_atlas.position * new_texture_region_size + extra;
```

Comment:
```suggestion
					Vector2i new_position = new_tile_rect_in_atlas.position * new_texture_region_size + extra;
```
Style requires snake case